src/Microsoft.AspNetCore.Mvc.Razor.Extensions/ViewComponentTagHelperPass.cs

```diff
@@ -71,8 +72,8 @@ private void GenerateVCTHClass(ClassDeclarationIRNode @class, TagHelperDescripto
             var newTypeName = GetVCTHFullName(@namespace, @class, node.Descriptor);
             for (var i = 0; i < node.Parent.Children.Count; i++)
             {
-                var setProperty = node.Parent.Children[i] as SetTagHelperPropertyIRNode;
-                if (setProperty != null)
+                if (node.Parent.Children[i] is SetTagHelperPropertyIRNode setProperty &&
+                    string.Equals(originalTypeName, setProperty.TagHelperTypeName, StringComparison.Ordinal))
```

Comment:
Yea both `CreateTagHelperIRNode` and `SetTagHelperPropertyIRNode` have descriptors. But not sure what you mean by use them.